grails-app/services/au/org/ala/profile/hub/ImageService.groovy

```diff
@@ -286,16 +286,42 @@ class ImageService {
+        Integer numberOfIncludedLocalImages = 0;
+
+
+        switch (opusDefaultOption) {
+            case ImageOption.INCLUDE:
+                List excluded = profile?.imageSettings.findAll { isExcluded(opusDefaultOption, it?.displayOption?.toString()) }*.imageId
+                //find all private images from profile imageSettings with same imageId from profile privateImages because there is no private info in imageSettings
+                List privateImages = profile?.imageSettings.findAll {profile?.privateImages?.imageId.contains(it?.imageId?.toString())}
+                if (privateImages?.size() > 0) {
+                    List excludedPrivateImages = privateImages?.findAll { isExcluded(opusDefaultOption, it?.displayOption?.toString()) }*.imageId
+                    if (excludedPrivateImages?.size() > 0) {
+                        numberOfIncludedLocalImages = privateImages.size() - excludedPrivateImages.size()
+                    } else {
+                        numberOfIncludedLocalImages = privateImages.size()
+                    }
+                }
 
-        List excluded = profile?.imageSettings.findAll { isExcluded(opusDefaultOption, it?.displayOption?.toString()) }*.imageId
-
-        if (readonlyView) {
-            minusQuery = excluded.collect { "-image_url: $it" }.join(' AND ')
+                if ((excluded?.size() > 0) && readonlyView) {
+                    includeExcludeQuery = excluded.collect { "-image_url:$it" }.join(' AND ')
+                }
+                break
+            case ImageOption.EXCLUDE:
+                List included = profile?.imageSettings.findAll { isIncluded(opusDefaultOption, it?.displayOption?.toString()) }*.imageId
+                //find all private images from profile imageSettings with same imageId from profile privateImages because there is no private info in imageSettings
+                List privateImages = profile?.imageSettings.findAll {profile?.privateImages?.imageId.contains(it?.imageId?.toString())}
+                List includedPrivateImages = privateImages?.findAll { isIncluded(opusDefaultOption, it?.displayOption?.toString()) }*.imageId
+                numberOfIncludedLocalImages = (includedPrivateImages)? includedPrivateImages.size() : 0
+                if ((included?.size() > 0) && readonlyView) {
+                    includeExcludeQuery = "(" + included.collect { "image_url:$it" }.join(' OR ') + ")"
+                }
+                break
         }
 
-        Map numberOfPublishedImagesMap = biocacheService.imageCount(searchIdentifier, opus, minusQuery)
+        Map numberOfPublishedImagesMap = biocacheService.imageCount(searchIdentifier, opus, includeExcludeQuery)
         if (numberOfPublishedImagesMap && numberOfPublishedImagesMap?.resp && numberOfPublishedImagesMap?.resp?.totalRecords > 0) {
             numberOfPublishedImages = numberOfPublishedImagesMap?.resp?.totalRecords
         }
```

Comment:
update privateImages code to make sure private images are returned based on read only or write mode.